### PR TITLE
Docs: Consistent dev environment

### PR DESCRIPTION
**Pull request by travisn**
Signed-off-by: travisn <tnielsen@redhat.com>

**Description of your changes:**
Add more details on building and setting up a dev environment to help onboarding of new contributors.

**Checklist:**
- [ ] Documentation has been updated, if necessary.
- [ ] Pending release notes updated with breaking and/or notable changes, if necessary.
- [ ] Upgrade from previous release is tested and upgrade user guide is updated, if necessary.
- [ ] Code generation (`make codegen`) has been run to update object specifications, if necessary.
- [ ] Comments have been added or updated based on the standards set in [CONTRIBUTING.md](../blob/master/CONTRIBUTING.md#comments)

[skip ci]

### CODE REVIEW

File: Documentation/development-flow.md [hunk 1]
```diff
@@ -33,11 +33,41 @@ cd $GOPATH/src/github.com/rook
...
+the following settings in your IDE:
+- Format with the `goreturns` tool
+- Trim trailing whitespace
+
+For example, in VS Code this translates to the following settings:
+
+```json
+{
+    "editor.formatOnSave": true,
+    "go.buildOnSave": true,
```
Comment (jbw976) [inline Documentation/development-flow.md]: mine doesn't list `true` as an option, using VS Code 1.30.1.  I wonder if that's a stale comment that's been in the settings file for a long time?
```
 // Compiles code on file save using 'go build -i' or 'go test -c -i'. Options are 'workspace', 'package or 'off'.
  "go.buildOnSave": "package"
```

File: Documentation/development-flow.md [hunk 2]
```diff
@@ -33,11 +33,41 @@ cd $GOPATH/src/github.com/rook
...
+    "go.formatTool": "goreturns",
+    "files.trimTrailingWhitespace": true,
+    "files.insertFinalNewline": true,
```
Comment (jbw976) [inline Documentation/development-flow.md]: oh i like this one to insert a final new line!

File: Documentation/development-flow.md (hunk 1 above)
Comment (BlaineEXE) [reply Documentation/development-flow.md]: I would suggest `"package"` as the setting here. Rebuilding the entirety of Rook on save is resource intensive. Though the recommendations could be `"go.buildOnSave": "package" // or "workspace" if your machine builds Rook quickly

**Comment by BlaineEXE** [comment]
Love it. <3

How do you feel about adding a small section for how to run a subset of Unit and Integration tests?

File: Documentation/development-flow.md (hunk 2 above)
Comment (travisn) [reply Documentation/development-flow.md]: yes, thanks to @galexrt for that suggestion!

File: Documentation/development-flow.md (hunk 1 above)
Comment (travisn) [reply Documentation/development-flow.md]: Yep, must be a very old setting. Agreed that "package" works well.

**Comment by travisn** [comment]
@BlaineEXE There is a section lower in the doc on unit and integration tests. You're suggesting to add more examples how to execute them for an individual package or suite?

File: Documentation/development-flow.md (hunk 2 above)
Comment (leseb) [reply Documentation/development-flow.md]: AFAIK all of that is handled automatically by the Go extension. My `files.insertFinalNewline": true` is not set and I get the new empty line anyways.

**Pull request merged**